tests/phpunit/tests/menu/wp-nav-menu.php

```diff
@@ -197,4 +197,52 @@ public function test_wp_nav_menu_should_not_have_has_children_class_with_custom_
 			'Level 3 should not be present in the HTML output.'
 		);
 	}
+
+	/**
+	 * Test that the order in which menu items are created does not matter.
+	 *
+	 * @ticket 57122
+	 */
+	public function test_parent_with_higher_id_should_not_error() {
```

Comment:
Just to confirm, the parent of `self::$lvl1_menu_item` does not need to be reset to `self::$lvl0_menu_item` as the parent `set_up()/tear_down()` methods handle resetting this. Is that correct?

All other tests pass if you move this new test to the top of the file, and this drop-in test method also passes:
```php
public function test_that_peter_did_not_break_the_test_suite() {
	$menu_items = wp_get_nav_menu_items( self::$menu_id );
	$this->assertSame( (int) $menu_items[1]->menu_item_parent, self::$lvl0_menu_item );
}
```

But just asking to be sure.

Reply:
> Just to confirm, the parent of `self::$lvl1_menu_item` does not need to be reset to `self::$lvl0_menu_item` as the parent `set_up()/tear_down()` methods handle resetting this. Is that correct?

Yes, that is correct.

Changes to the post and post meta table are stored in a transaction that is started in the test suite's setup and rolled back during the tear-down. 

https://github.com/WordPress/wordpress-develop/blob/389a78ecc453036f144b6d510cc2423f41bc9dd1/tests/phpunit/includes/abstract-testcase.php#L136-L146